From 8288740d85837548b758cdfe62756c3967c00aab Mon Sep 17 00:00:00 2001
From: Sultan Alsawaf <sultan@kerneltoast.com>
Date: Mon, 21 Aug 2023 21:34:02 -0700
Subject: [PATCH] wifi: iwlwifi: mvm: allow DTIM periods to be skipped on radar
 channels

The reason for disallowing DTIM periods from being skipped on radar
channels is unclear, and originates from the very introduction of the MVM
driver. Since it hurts power consumption with no evident good reason,
remove it to allow DTIM periods to be skipped on radar channels.

Signed-off-by: Sultan Alsawaf <sultan@kerneltoast.com>
---
 drivers/net/wireless/intel/iwlwifi/mvm/power.c | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/power.c b/drivers/net/wireless/intel/iwlwifi/mvm/power.c
index bc363e8427e4..ac233ffdbd61 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/power.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/power.c
@@ -282,19 +282,6 @@ static bool iwl_mvm_power_allow_uapsd(struct iwl_mvm *mvm,
 	return data.allow_uapsd;
 }
 
-static bool iwl_mvm_power_is_radar(struct ieee80211_bss_conf *link_conf)
-{
-	struct ieee80211_chanctx_conf *chanctx_conf;
-
-	chanctx_conf = rcu_dereference(link_conf->chanctx_conf);
-
-	/* this happens on link switching, just ignore inactive ones */
-	if (!chanctx_conf)
-		return false;
-
-	return chanctx_conf->def.chan->flags & IEEE80211_CHAN_RADAR;
-}
-
 static void iwl_mvm_power_config_skip_dtim(struct iwl_mvm *mvm,
 					   struct ieee80211_vif *vif,
 					   struct iwl_mac_power_cmd *cmd)
@@ -321,7 +308,7 @@ static void iwl_mvm_power_config_skip_dtim(struct iwl_mvm *mvm,
 		unsigned int dtimper_tu = dtimper * link_conf->beacon_int;
 		unsigned int skip;
 
-		if (dtimper >= 10 || iwl_mvm_power_is_radar(link_conf)) {
+		if (dtimper >= 10) {
 			rcu_read_unlock();
 			return;
 		}

